libr/util/str_trim.c

```diff
@@ -101,81 +105,62 @@ R_API void r_str_trim(char *str) {
 			*ptr = '\0';
 		}
 	}
+#endif
 }
 
 // Returns a pointer to the first non-whitespace character of str.
-// TODO: rename to r_str_trim_head_ro()
-R_API const char *r_str_trim_ro(const char *str) {
-	if (str) {
-		for (; *str && IS_WHITECHAR (*str); str++) {
-			;
-		}
-	}
-	return str;
-}
-
-// Returns a pointer to the first whitespace character of str.
-// TODO: rename to r_str_trim_head_wp()
-R_API const char *r_str_trim_wp(const char *str) {
-	if (str) {
-		for (; *str && !IS_WHITESPACE (*str); str++) {
-			;
-		}
+// TODO: Find a better name: to r_str_trim_head_ro(), r_str_skip_head or so
+R_API const char *r_str_trim_head_ro(const char *str) {
+	r_return_val_if_fail (str, NULL);
+	for (; *str && IS_WHITECHAR (*str); str++) {
+		;
 	}
 	return str;
 }
 
 /* remove spaces from the head of the string.
  * the string is changed in place */
-R_API char *r_str_trim_head(char *str) {
-	char *p = (char *)r_str_trim_ro (str);;
+R_API void r_str_trim_head(char *str) {
+	char *p = (char *)r_str_trim_head_ro (str);;
 	if (p) {
 		memmove (str, p, strlen (p) + 1);
 	}
-	return str;
 }
 
 // Remove whitespace chars from the tail of the string, replacing them with
 // null bytes. The string is changed in-place.
-R_API char *r_str_trim_tail(char *str) {
-	if (!str) {
-		return NULL;
-	}
+R_API void r_str_trim_tail(char *str) {
+	r_return_if_fail (str);
 	size_t length = strlen (str);
-	if (!length) {
-		return str;
-	}
-
-	while (length--) {
+	while (length-- > 0) {
 		if (IS_WHITECHAR (str[length])) {
 			str[length] = '\0';
 		} else {
 			break;
 		}
 	}
-
-	return str;
 }
 
 // Removes spaces from the head of the string, and zeros out whitespaces from
 // the tail of the string. The string is changed in place.
-R_API char *r_str_trim_head_tail(char *str) {
-	return r_str_trim_tail (r_str_trim_head (str));
+R_API void r_str_trim_head_tail(char *str) {
+	r_str_trim_head (str);
+	r_str_trim_tail (str);
 }
 
-// no copy, like trim_head+tail but with trim_head_ro
+// no copy, like trim_head+tail but with trim_head_ro, beware heap issues
+// TODO: rename to r_str_trim_weak() ?
 R_API char *r_str_trim_nc(char *str) {
```

Comment:
what does `nc` stand for?

Reply:
no-copy, its a shitty name but i didnt decided for it, and i think its worth having it for performance reasons